PF/Assignment 4/Day4Assignment.java

```diff
@@ -0,0 +1,96 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Day4Assignment
+{
+	final int NUMBER_OF_COLUMNS = 4; // for matrix of 4X4 for N Queen Problem
+	List<String> towerStrings = new ArrayList<String>();
+
+	// function to check for TOwer of Hanoi solution
+	public List<String> towerOfHanoi(String source, String destination, String temp, int numOfDisk)
+	{
+		if (numOfDisk > 1)
+		{
+			// Move top n-1 disks from source to destination, using temp as auxiliary
+			towerOfHanoi(source, temp, destination, numOfDisk - 1);
+			// Move remaining disks from source to temp
+			towerStrings.add("Move Disk " + numOfDisk + " from " + source + " to " + destination);
+			// Move n-1 disks from destination to temp using source as auxiliary
+			towerOfHanoi(temp, destination, source, numOfDisk - 1);
+		}
+		else // If only 1 disk, make the move and return
+		if (numOfDisk == 1)
+		{
+			towerStrings.add("Move Disk " + numOfDisk + " from " + source + " to " + destination);
+			return towerStrings;
+		}
+		return towerStrings;
+	}
+
+	// function to check for permutation solution
+	public List<String> generatePermutations(String input, String str)
+	{
+		List<String> permutations = new ArrayList<>();
+		int n = str.length();
+		if (n == 0)
+			permutations.add(input);
+		else
+			for (int i = 0; i < n; i++)
```

Comment:
if there is loop in block, enclose it in curly braces even there is single loop.